### PR TITLE
Update externals definition for Webpack V4

### PR DESCRIPTION
**Pull request by rwaskiewicz**
Change:
- Externals should accept objects as their values (see https://webpack.js.org/configuration/externals/\#object);

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://webpack.js.org/configuration/externals/#object
- [X] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/webpack/index.d.ts [hunk 1]
```diff
@@ -350,7 +351,7 @@ declare namespace webpack {
     type ExternalsElement = string | RegExp | ExternalsObjectElement | ExternalsFunctionElement;
 
     interface ExternalsObjectElement {
-        [key: string]: boolean | string;
+        [key: string]: boolean | string | object;
```
Comment (Kovensky) [inline types/webpack/index.d.ts]: this `object` is specifically a `Record<string, string>`, though.

Comment (johnnyreilly) [reply types/webpack/index.d.ts]: @rwaskiewicz could you make the amend suggested please?

Comment (mohsen1) [reply types/webpack/index.d.ts]: [In one of the tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b23d51e8daedc06fd41b88304b56577f0f396ed2/types/webpack/webpack-tests.ts#L104) it's mapped to `string[]` or am I missing something?

Comment (johnnyreilly) [reply types/webpack/index.d.ts]: Hmmm. Don't know.

**Review by mohsen1 — APPROVED** [review]

**Comment by johnnyreilly** [comment]
Thanks!

**Pull request merged**